code/_onclick/adjacent.dm

```diff
@@ -29,10 +29,10 @@
 	var/turf/T0 = get_turf(neighbor)
 
 	if(T0 == src) //same turf
-		return 1
+		return TRUE
 
-	if(get_dist(src,T0) > 1) //too far
-		return 0
+	if(get_dist(src, T0) > 1 || z != T0.z) //too far

```

Comment:
just ```in_range()``` surely?

Reply:
In theory there is some tiny tiny amount of perf saved by abusing the knowledge that `src` and `T0` are turfs and we can use `z` directly but yes, it should be equivalent to `in_range`.